app/models/accounting/interest_calculator.rb

```diff
@@ -205,10 +234,20 @@ def month_boundaries(d1, d2)
     end
 
     def add_int_tx?(txs, prev_tx)
-      return true if txs.nil?
-      if txs
-       return true if prev_tx && prev_tx.principal_balance > 0 && txs.none?(&:interest?)
+      if txs.nil? # this is an end of month day with no txns
```

Comment:
This method seems to have too much nesting. I think we will want to know the difference between "no end of month int txn before closed books date" vs "there was a txn that wasn't paired with an interest txn before closed books date." I think the "no end of month int txn" could get really annoying and we could remove it if so; need to try it out with real data and see how common it is.